lib/core/hard_machine.rb

```diff
@@ -0,0 +1,61 @@
+require "core/board"
+require "core/player"
+
+# HardMachine class follows the minmax algorithm to choose the next move.
+# The algorithm has been implemented taking into account depth in order
+# to be able to identify quicker wins over longer ones.
+class HardMachine < Player
+  INITIAL_DEPTH = 0
+
+  def initialize(symbol)
+    super(symbol)
+    @opponent = Player.new(symbol == "X" ? "O" : "X")
+  end
+
+  def get_move(board)
+    @current_player = self
+    minimax(board, INITIAL_DEPTH)
+    @choice
+  end
+
+  private
+
+  def minimax(board, depth)
+    return score(board, depth) if board.complete? || board.won?(self) || board.won?(@opponent)
```

Comment:
The predicate for the conditional would read a lot better if you extracted those three things into a method or the board. i.e., if game_over?